arbnode/message_pruner_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package arbnode
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/ethdb"
+)
+
+func TestMessagePrunerWithOldMessagesPresent(t *testing.T) {
+	endBatchCount := uint64(3)
+	endBatchMetadata := BatchMetadata{
+		MessageCount:        3,
+		DelayedMessageCount: 3,
+	}
+	inboxTrackerDb, transactionStreamerDb := setupDatabase(t, endBatchCount, endBatchMetadata)
+	deleteOldMessageFromDB(endBatchCount, endBatchMetadata, inboxTrackerDb, transactionStreamerDb)
+
+	checkDbKeys(t, endBatchCount, inboxTrackerDb, sequencerBatchMetaPrefix)
+	checkDbKeys(t, uint64(endBatchMetadata.MessageCount), transactionStreamerDb, messagePrefix)
+	checkDbKeys(t, endBatchMetadata.DelayedMessageCount, inboxTrackerDb, rlpDelayedMessagePrefix)
+
+}
+
+func TestMessagePrunerWithOldMessagesNotPresent(t *testing.T) {
+	endBatchCount := uint64(2)
+	endBatchMetadata := BatchMetadata{
+		MessageCount:        2,
+		DelayedMessageCount: 2,
+	}
+	inboxTrackerDb, transactionStreamerDb := setupDatabase(t, endBatchCount, endBatchMetadata)
+	deleteOldMessageFromDB(endBatchCount, endBatchMetadata, inboxTrackerDb, transactionStreamerDb)
+
+	checkDbKeys(t, endBatchCount, inboxTrackerDb, sequencerBatchMetaPrefix)
+	checkDbKeys(t, uint64(endBatchMetadata.MessageCount), transactionStreamerDb, messagePrefix)
+	checkDbKeys(t, endBatchMetadata.DelayedMessageCount, inboxTrackerDb, rlpDelayedMessagePrefix)
+
+}
+
```

Comment:
we probably want a test that goes over many more keys.. enough to trigger the "if batch.ValueSize() >= ethdb.IdealBatchSize" in deleteFromRange.

Reply:
done.